### ISSUE TITLE
Do not warn about from_over_into with external crate types

### ISSUE THREAD

**Issue by almindor** [rust-lang/rust-clippy#7444]
Lint name: `from_over_into`

This lint should not be attempted on types that live in external crates. Right now it triggers e.g.:
```rust
impl Into<OutOfCrateType> for InCrateType {...}
```

I expected to see this happen: 
*Not to get this case linted as a warning because the `OutOfCrateType` is external and thus `From` cannot or might not be implementable on it since `InCrateType` is from this crate.*

Instead, this happened: 
*It got linted as a warning.*

### Meta

- `cargo clippy -V`: clippy 0.1.51 (2fd73fa 2021-03-23)
- `rustc -Vv`:
```
rustc 1.51.0 (2fd73fabe 2021-03-23)
binary: rustc
commit-hash: 2fd73fabe469357a12c2c974c140f67e7cdd76d0
commit-date: 2021-03-23
host: x86_64-unknown-linux-gnu
release: 1.51.0
LLVM version: 11.0.1
```

**Comment by oribs1** [comment]
After a short research, the restriction was weakened in Rust 1.41.0 to allow these implementations:
https://blog.rust-lang.org/2020/01/30/Rust-1.41.0.html

So this lint is ok in the case of Into
